### PR TITLE
Merging Euler to traj_opt

### PR DESCRIPTION
**Pull request by somritabanerjee**

### CODE REVIEW

File: mobility/planner_scp/scp/src/traj_opt.cpp [hunk 1]
```diff
@@ -316,17 +856,34 @@ void TOP::UpdateF(Vec7& f, Vec13& X, Vec6& U) {
...
+                -wx, -wy, -wz, 0;
+  A.block(0, 0, 4, 4) = 0.5*df_dq;
+
+  // TODO(somrita): Remove below matrix
+  // omega_skew << 0, wz, -wy, wz,
+  //           -wz, 0, wx, wy,
+  //           wy, -wx, 0, wz,
+  //           -wx, -wy, -wz, 0;
+
+  Mat4x3 df_dw;
```
Comment (acauligi) [inline mobility/planner_scp/scp/src/traj_opt.cpp:881]: `Mat4x3` doesn't seem to be defined in _types.h_ .

File: mobility/planner_scp/scp/src/traj_opt.cpp [hunk 2]
```diff
@@ -360,872 +914,1150 @@ void TOP::UpdateDynamics() {
...
+//       }
+//       nhat.normalize();
+//       obs_ub[n_obs*ii+jj] = sd_result.sd - nhat.dot(Xprev[ii].segment(0, 3)) - obs_clearance;
+//       support_vectors[n_obs*ii+jj] = -nhat;
+//     }
+//   }
+// }
 
-    row_idx += 3;
+// void TOP::SetHessianMatrix() {
```
Comment (acauligi) [inline mobility/planner_scp/scp/src/traj_opt.cpp:1300]: Shall we go ahead and delete all the `Set{}Constraints` and methods from the previous SCP architecture if we won't be switching back to them?

File: mobility/planner_scp/scp/src/traj_opt.cpp (hunk 1 above)
Comment (somritabanerjee) [reply mobility/planner_scp/scp/src/traj_opt.cpp:881]: Eagle eye! Yes, I forgot to copy over types.h. I added a commit to this branch.

File: mobility/planner_scp/scp/src/traj_opt.cpp (hunk 2 above)
Comment (somritabanerjee) [reply mobility/planner_scp/scp/src/traj_opt.cpp:1300]: I admit this is a personal crutch, but it really helps me after writing a piece of constraints code, to go back and check your earlier implementation. I assume the probability of us both getting it right is greater than the probability of us both getting it wrong independently. Of course, I could do this by just saving your earlier code in a separate file. I'm ok with deleting it. 

In the future though, I think going back to the modular function design is a good idea. For now, it makes testing easier to have it in one function.

**Pull request merged**